azurerm/internal/services/containers/container_registry_resource.go

```diff
@@ -89,12 +88,6 @@ func resourceContainerRegistry() *schema.Resource {
 				Default:  true,
 			},
 
-			"storage_account_id": {
```

Comment:
same here, i don't see any reason to remove it now instead of deprecating it and removing in 3.0